src/node_buffer.cc

```diff
@@ -499,6 +530,96 @@ void ReadDoubleBE(const FunctionCallbackInfo<Value>& args) {
 }
 
 
+template <typename T, enum Endianness endianness, T min, T max>
+void ReadInt64Generic(const FunctionCallbackInfo<Value>& args, const char *formatter) {
+  Environment* env = Environment::GetCurrent(args);
+
+  ARGS_THIS(args[0].As<Object>());
+
+  uint32_t offset = args[1]->Uint32Value();
+  CHECK_LE(offset + sizeof(T), obj_length);
+
+  union NoAlias {
+    T val;
+    char bytes[sizeof(T)];
+  };
+
+  union NoAlias na;
+  const char* ptr = static_cast<const char*>(obj_data) + offset;
+  memcpy(na.bytes, ptr, sizeof(na.bytes));
+  if (endianness != GetEndianness())
+    Swizzle(na.bytes, sizeof(na.bytes));
+
+  if (na.val < min || na.val > max) {
+    // return a String
+    char strbuf[20];
+    int len = snprintf(strbuf, sizeof(strbuf), formatter, na.val);
+    args.GetReturnValue().Set( node::OneByteString(env->isolate(), strbuf, len) );
+  } else {
+    // return a Number
+    args.GetReturnValue().Set(static_cast<double>(na.val));
+  }
+}
+
+
+void ReadInt64LE(const FunctionCallbackInfo<Value>& args) {
+  ReadInt64Generic<int64_t, kLittleEndian, JS_MIN_INT, JS_MAX_INT>(args, "%" PRId64);
+}
+
+
+void ReadInt64BE(const FunctionCallbackInfo<Value>& args) {
+  ReadInt64Generic<int64_t, kBigEndian, JS_MIN_INT, JS_MAX_INT>(args, "%" PRId64);
+}
+
+
+void ReadUInt64LE(const FunctionCallbackInfo<Value>& args) {
+  ReadInt64Generic<uint64_t, kLittleEndian, 0, JS_MAX_INT>(args, "%" PRIu64);
+}
+
+
+void ReadUInt64BE(const FunctionCallbackInfo<Value>& args) {
+  ReadInt64Generic<uint64_t, kBigEndian, 0, JS_MAX_INT>(args, "%" PRIu64);
+}
+
+
+template <enum Endianness endianness>
+void ReadPointerGeneric(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+  EscapableHandleScope scope(env->isolate());
+
+  ARGS_THIS(args[0].As<Object>());
+
+  uint32_t offset = args[1]->Uint32Value();
+  CHECK_LE(offset + sizeof(char *), obj_length);
+
+  size_t size = args[2]->Uint32Value();
+
+  char* ptr = static_cast<char*>(obj_data) + offset;
+
+  union NoAlias {
+    char* val;
+    char bytes[sizeof(char**)];
+  };
+
+  union NoAlias na = { *reinterpret_cast<char **>(ptr) };
+
+  if (endianness != GetEndianness())
+    Swizzle(na.bytes, sizeof(na.bytes));
+
+  args.GetReturnValue().Set( scope.Escape( Buffer::Use(env, na.val, size) ) );
```

Comment:
We're basically saying "OK, read this pointer, now make a Buffer at that address. Now how much data at that point in memory at you interested in looking at?".

So if we knew that the data at the memory address we are reading is an `int`, then we would set the size to 4 bytes so that we could execute a `readInt32LE(0)` call afterwards to "unbox".